src/bin/export.js

```diff
@@ -1,264 +1,262 @@
+ * @author brollb / https://github.com/brollb
+ */
+'use strict';
+var webgme = require('../../webgme'),
+    FS = require('fs'),
+    Q = require('q'),
+    cliStorage,
+    gmeAuth,
+    blobClient,
+    MongoURI = require('mongo-uri'),
+    path = require('path'),
+    gmeConfig,
+    logger,
+    Serialization = webgme.serializer,
+    STORAGE_CONSTANTS = webgme.requirejs('common/storage/constants'),
+    REGEXP = webgme.requirejs('common/regexp'),
+    main,
+    mainDeferred,
+    runInternal,
+    runRaw,
+    run,
+    getMissingRequiredParam,
+    BlobClient = require('../../src/server/middleware/blob/BlobClientWithFSBackend'),
+    storageUtils = webgme.requirejs('common/storage/util'),
+    blobUtil = webgme.requirejs('common/blob/util');
+
+/**
+ * Check for missing required parameters
+ *
+ * @param {Object} params
+ * @return {String|null} missing param
+ */
+getMissingRequiredParam = function (params) {
+    var requiredParams = ['projectName', 'source', 'outFile'];
+    for (var i = requiredParams.length; i--;) {
+        if (!params[requiredParams[i]]) {
+            return requiredParams[i];
+        }
+    }
+    return null;
+};
+
+/**
+ * Entrypoint for CLI usage
+ *
+ * @param {Array<String>} argv
+ * @return {undefined}
+ */
+main = function (argv) {
+    var Command = require('commander').Command,
+        program = new Command(),
+        syntaxFailure = false;
+
+    gmeConfig = require(path.join(process.cwd(), 'config'));
+    logger = webgme.Logger.create('gme:bin:export', gmeConfig.bin.log, false);
+    blobClient = new BlobClient(gmeConfig, logger.fork('BlobClient'));
+    webgme.addToRequireJsPaths(gmeConfig);
+    mainDeferred = Q.defer();
+    program
+        .version('0.2.0')
+        .option('-m, --mongo-database-uri [url]',
+            'URI of the MongoDB [by default we use the one from the configuration file]')
+        .option('-u, --user [string]', 'the user of the command [if not given we use the default user]')
+        .option('-p, --project-name [string]', 'project name [mandatory]')
+        .option('-o, --owner [string]', 'the owner of the project [by default, the user is the owner]')
+        .option('-s, --source [branch/commit]', 'the branch or commit that should be exported')
+        .option('-f, --out-file [path][mandatory]', 'the path of the output file')
+        .option('-n, --no-assets', 'if given, no assets will be bundled into the package')
+        .parse(argv);
+
+    if (getMissingRequiredParam(program)) {
+        logger.error(getMissingRequiredParam(program) + ' is a mandatory parameter!');
+        syntaxFailure = true;
+    }
+
+    if (syntaxFailure) {
+        program.outputHelp();
+        mainDeferred.reject(new SyntaxError('invalid argument'));
+        return mainDeferred.promise;
+    }
+    return runInternal(program);
+};
+
+/**
+ * Entry point for usage as a module.
+ *
+ * @param {Dictionary<String,String>} params
+ * @return {undefined}
+ */
+run = function (params) {
```

Comment:
noted. 
I removed run and runRaw as they are not needed anymore.
We could also clear out runInternal, but for now it can stay and it uses the `function runInternal(` form of definition.